File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/search/model/multi/SearchMultiResponse.java
package dev.reuss.tmdb.domain.search.model.multi;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.common.page.PagedResponse;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Paginated TMDB multi search response.
 *
 * @param page         current result page
 * @param results      found movies, TV shows and people
 * @param totalPages   total available pages
 * @param totalResults total available results
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record SearchMultiResponse(
        int page,

        List<SearchMultiResult> results,

        @JsonProperty("total_pages")
        int totalPages,

        @JsonProperty("total_results")
        int totalResults
) implements PagedResponse<SearchMultiResult>, TmdbModel {

    public SearchMultiResponse {
        results = TmdbCollections.list(results);
        ;
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/search/model/multi/SearchMultiResponse.kt
package dev.reuss.tmdb.domain.search.model.multi

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel
import dev.reuss.tmdb.common.page.PagedResponse

/**
 * Paginated TMDB multi search response.
 *
 * @property page current result page
 * @property results found movies, TV shows and people
 * @property totalPages total available pages
 * @property totalResults total available results
 */
@JvmRecord
data class SearchMultiResponse(
    @all:JsonProperty("page")
    override val page: Int,
    @all:JsonProperty("results")
    override val results: List<SearchMultiResult> = emptyList(),
    @all:JsonProperty("total_pages")
    override val totalPages: Int,
    @all:JsonProperty("total_results")
    override val totalResults: Int,
) : PagedResponse<SearchMultiResult>,
    TmdbModel
